### DISCUSSION TITLE
Docker Volumes.

### DISCUSSION THREAD

**Discussion by CrazyHenk44** [photoprism/photoprism#3682] [Ideas]
On https://docs.photoprism.app/getting-started/docker-compose/ there is a bit about volumes and the syntax in the stack yaml is kind of non standard. I was wondering if it would be an idea to use the official volumes and that probably gets rid of the OS specific bits as well (not tested).

Like this:

![Safari Portainer  local 19 31 43@2x](https://github.com/photoprism/photoprism/assets/5298235/42a7844d-15a8-4178-85d6-e73933d745d2)

**Comment by lastzero** [comment]
You generally want to mount existing files. That's not possible with named volumes who don't reference a host folder. Also it's more difficult to understand if you need to edit the same volumes in two sections, one on the top and one at the bottom. How is mounting an existing folder non standard?

**Comment by CrazyHenk44** [comment]
Well, I did not really think while using the de suggested composer file and ended up with folders in /data/compose/2, which was outside the normal docker volume location, hence it was not included in my backups.  With this, it uses the default /var/lib/docker/volumes method, just like for example, WordPress: https://github.com/docker/awesome-compose/tree/master/official-documentation-samples/wordpress/

**Reply by lastzero** [reply]
It's unlikely that your existing photo library is already stored in /var/lib/docker at the time you customize the configuration, no?

**Comment by CrazyHenk44** [comment]
I think most people want to start with an empty folder and then copy in all images from all the different locations.
It was just an idea, thanks for considering anyway!

**Reply by lastzero** [reply]
If you want to use Docker volumes only, without configuring a separate path for each folder, you can simplify the config as follows:

```yaml
services:
  photoprism:
    ...
    volumes:
      - /photoprism
 ```

Or with a named volume:
```yaml
services:
  photoprism:
    ...
    volumes:
      - photoprism_files:/photoprism
 
volumes:
  photoprism_files:
```
